pep-XXXX.rst

```diff
@@ -0,0 +1,492 @@
+
+
+Syntax Closer to Function Signatures
+------------------------------------
+
+One alternative we had floated was a syntax much more similar to function signatures.
+
+In this proposal, the following types would have been equivalent::
+
+    class Function(typing.Protocol):
+        def f(self, x: int, /, y: float, *, z: bool = ..., **kwargs: str) -> bool:
+            ...
+
+    Function = (x: int, /, y: float, *, z: bool = ..., **kwargs: str) -> bool
+
+
+The benefits of this proposal would have included:
+- Perfect syntactic consistency between signatures and callable types.
+- Support for more features of function signatures (named, optional, variadic args) that this PEP does not support.
+
+Key downsides that led us to reject the idea include the following:
+- A large majority of use cases only use positional-only arguments, and this syntax would be more verbose for that use case, both because of requiring argument names and an explicit ``/``, for example ``(int, /) -> bool`` where our proposal allows ``(int) -> bool``
+- The requirement for explicit ``/`` for positional-only arguments has a high risk of causing frequent bugs - which often wouldn’t be detected by unit tests - where library authors would accidentally use types with named arguments.
+- Our analysis suggests that support for ``ParamSpec`` is key, but the scope rules laid out in PEP 612 would have made this difficult.
+
+
+Other Proposals Considered
+--------------------------
+
+An idea we looked at very early on was to allow using functions as types. This may be a great idea, but we consider less an alternative to better callable types than a major improvement in the usability of Callable Protocols:
+- Using functions as types wouldn’t give us a new way of describing function types as first class values. Instead, they would require a function definition statement that effectively defines a type alias (much as a Callable Protocol class statement does).
+- Functions-as-types would support almost exactly the same features that Callable Protocols do today: named, optional, and variadic args as well as the ability to define overloads.
+So we think that is an idea for a related PEP, but not a direct substitute for improved Callable syntax.
+
+We considered a parentheses-free syntax that would have been even more concise::
+
+    int, str -> bool
+
+We decided against it because this is not visually as similar to existing function header syntax. Moreover, it is visually similar to lambdas, which bind names with no parentheses: ``lambda x, y: x == y``.
+
+Another idea was a new “special string” syntax an puting the type inside of it, for example ``t”(int, str) -> bool”``. We rejected this because it is not as readable, and it doesn’t seem in line with guidance from the Steering Council on ensuring that type expressions do not diverge from the rest of Python syntax. [#python-types-and-runtime-guidance]_
```

Comment:
```suggestion
Another idea was a new “special string” syntax an puting the type inside of it, for example ``t”(int, str) -> bool”``. We rejected this because it is not as readable, and seems out of step with guidance from the Steering Council on ensuring that type expressions do not diverge from the rest of Python's syntax. [#python-types-and-runtime-guidance]_
```